src/mumble/GlobalShortcut_win.cpp

```diff
@@ -433,24 +456,26 @@ void GlobalShortcutWin::injectRawInputMessage(HRAWINPUT handle) {
 				return;
 			}
 
-			emit keyboardMessage(keyboard.Flags, keyboard.MakeCode, keyboard.VKey);
+			m_msgQueue.emplace(std::make_unique< MsgKeyboard >(keyboard.Flags, keyboard.MakeCode, keyboard.VKey));
```

Comment:
From the SPSC docs for the `emplace` method:
> Enqueue an item using inplace construction. **Blocks if queue is full.**

I am wondering whether that blocking could potentially bite us in the back. If this code here (or any of the other emplaces) blocks, we never reach down to notifying the condition variable and are therefore dependent on the processing loop already running (consuming queue items and therefore allowing the emplacement to succeed once at least one item has been popped from the queue).
If for some reason this consumer loop was not running anyway, we'd be in an endless loop.

On the other hand these messages are added one at a time, right? Then after every single addition the condition variable is notified. Therefore, it should in theory be impossible for the consumer loop to not be running even though the queue is full :thinking:

Reply:
Correct.